.github/workflows/ci.yml

```diff
@@ -13,159 +13,103 @@ concurrency:
+              toolchain: gcc
+            - os: ubuntu-22.04
+              toolchain: msvc
+            - os: ubuntu-22.04
+              toolchain: clang
+            - os: windows-2022
+              toolchain: clang
+            - os: windows-2022
+              toolchain: gcc
+
+    name: CMake-${{ matrix.os }}-${{ matrix.build_type }}-${{ matrix.toolchain }}
     runs-on: ${{ matrix.os }}
+    container: ${{ matrix.container }}
+    env:
+      SCCACHE_GHA_ENABLED: "true"
 
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
 
-    - name: Set reusable strings
-      # Turn repeated input strings (such as the build output directory) into step outputs. These step outputs can be used throughout the workflow file.
-      id: strings
-      shell: bash
-      run: |
-        echo "build-output-dir=${{ github.workspace }}/out/build" >> "$GITHUB_OUTPUT"
-
-    - name: Install Ubuntu build dependencies
-      if: matrix.os == 'ubuntu-latest'
-      run: |
-        sudo apt update
-        sudo apt install libgl-dev libx11-xcb-dev libxrandr-dev libxinerama-dev libxcursor-dev libxi-dev libxext-dev wayland-protocols libwayland-dev libxkbcommon-dev
-
-    - name: Set up Python
-      uses: actions/setup-python@v4
-      with:
-        python-version: 3.11
-
-    - name: Fetch dependencies
-      run: python ./tools/fetch_dawn_dependencies.py --use-test-deps --shallow
-
-    - name: Set up Ninja
-      uses: seanmiddleditch/gha-setup-ninja@master
-
-    - name: Set up Visual Studio environment
-      if: matrix.os == 'windows-latest'
-      uses: seanmiddleditch/gha-setup-vsdevenv@master
+    - name: Set up dependencies on linux
+      if: matrix.container == 'dockcross/manylinux_2_28-x64:latest'
+      run: >
+        dnf install -y mesa-libGL-devel libxcb libxcb-devel libX11-xcb libXcursor-devel libXrandr-devel libXinerama-devel libXi-devel libXext-devel libxkbcommon libxkbcommon-devel libxkbcommon-x11-devel mesa-vulkan-drivers wayland-protocols-devel wayland-devel
 
     - name: Set up sccache
       uses: mozilla-actions/sccache-action@v0.0.3
 
     - name: Configure CMake
       run: >
-        cmake -G Ninja -B ${{ steps.strings.outputs.build-output-dir }}
+        cmake 
+        -S .
+        -B out/${{ matrix.build_type }}
         -DCMAKE_CXX_COMPILER=${{ matrix.cpp_compiler }}
         -DCMAKE_C_COMPILER=${{ matrix.c_compiler }}
         -DCMAKE_BUILD_TYPE=${{ matrix.build_type }}
         -DCMAKE_C_COMPILER_LAUNCHER=sccache
         -DCMAKE_CXX_COMPILER_LAUNCHER=sccache
-        -S ${{ github.workspace }}
-      env:
-        SCCACHE_GHA_ENABLED: "true"
+        -DDAWN_FETCH_DEPENDENCIES=ON
+        -DDAWN_ENABLE_INSTALL=ON
 
     - name: Build
-      run: cmake --build ${{ steps.strings.outputs.build-output-dir }} --config ${{ matrix.build_type }}
-      env:
-        SCCACHE_GHA_ENABLED: "true"
+      run: cmake --build out/${{ matrix.build_type }} --config ${{ matrix.build_type }} --target webgpu_dawn
```

Comment:
Maybe we could build all targets to get more coverage. WDYT?

Reply:
As the purpose of this code change was to package a dawn installation and the cmake code is written to only install `webgpu_dawn`, I thought it's a waste of machine time building extra targets, which won't make it to the package anyway. Considering the machines are free, it's probably okay, but still a waste of resources imo.

Reply:
I agree about building all targets since we want to keep other parts of the cmake build working as well.

Otherwise, looks great to me

Reply:
couple of errors when enabling all targets:

1. windows:

```
D:\a\dawn\dawn\src\tint\lang\hlsl\validate\validate.cc(53,10): error C1083: Cannot open include file: 'dxc/dxcapi.h': No such file or directory [D:\a\dawn\dawn\out\Debug\src\tint\tint_lang_hlsl_validate.vcxproj]
```

2. macos:

```
clang-15: error: no such file or directory: 'arm64'
clang-15: error: invalid Xarch argument: '-Xarch_arm64 -arch', options requiring arguments are unsupported
make[2]: *** [third_party/glslang/glslang/CMakeFiles/MachineIndependent.dir/MachineIndependent/glslang_tab.cpp.o] Error 1
make[1]: *** [third_party/glslang/glslang/CMakeFiles/MachineIndependent.dir/all] Error 2
make: *** [all] Error 2
```

I cannot reprod locally, so this has to do with toolchains/env. I'll look into it.

Reply:
why is glslang getting built on macos? I was under the assumption all opengl codepaths are disabled for macos.

Reply:
For dxc/dxcapi.h, I bet we need to add DXC to fetch_dawn_dependencies.py

Not sure about the Xarch one

Reply:
> For dxc/dxcapi.h, I bet we need to add DXC to fetch_dawn_dependencies.py

adding that will also need us to use `DAWN_USE_BUILT_DXC`, which increases build time. Shouldn't `dxcapi.h` already be available on the runner inside `%PROGRAMFILES(X86)%\Windows Kits\%SDKVERSION%\include\um` folder? I'll do some quick debugging later today.

Reply:
ah, that's `dxc/dxcapi.h` and not just `dxcapi.h`. it will need an update in `fetch_dawn_dependencies.py`